### DISCUSSION TITLE
Examples of modifying part of a file with chezmoi

### DISCUSSION THREAD

**Discussion by sheldonhull** [twpayne/chezmoi#1746] [Q&A] [answered]
I have an `.envrc` file with values in my home directory.
I want to make sure certain non-sensitive values are included with using a modify command, but haven't found any examples of how to properly use the modify command for this. Is it just a bash script that runs or can I use Go templating functions with conditional if exists type checks to add certain values if string isn't found using Sprig functions? If so I couldn't figure out what templating functions to use. 

I'll have to say chezmoi is one of the most exhaustively documented tools I've used, so much so that it's overwhelming at times to find the info :-) Might open a discussion sometime to look at Algolia full text integration for website since with so much info some better matching might help.

**Comment by twpayne** [comment, marked as answer]
`modify_` scripts can be anything that can be executed, including a bash script, a Python script, or even a compiled binary if you are so inclined.

If the `modify_` script has a `.tmpl` extension then chezmoi will interpret it as a template before executing the result.

If I understand your usecase correctly, you want to ensure that `.envrc` contains certain non-sensitive values. As these are non-sensitive, they can be included in plaintext in the `modify_` script itself so you do not need to use a template to extract secrets from a password manager.

Something like the following should work in `modify_dot_envrc`. Warning: I haven't tested this.
```bash
#!/bin/bash

# copy the data to modify to a temporary file
tmpfile=$(mktemp)
trap "rm -f ${tmpfile}" EXIT
cat > "${tmpfile}"

# write the data to stdout
cat "${tmpfile}"

# add any missing environment variables
if [ ! grep -q ^KEY= "${tmpfile}" ]; then
    echo KEY=VALUE
fi
```

**Comment by LorenzoBettini** [comment]
I think this is the correct syntax:

```bash
if ! grep -q ^KEY= "${tmpfile}"; then
```
